### PR TITLE
- DI-19703- Notifications - API Validation

### PR DESCRIPTION
**Pull request by NoamPepperi**
- DI-19711 - Notifications - Bulk API
- DI-19712 - Notifications Bulk API - API Validations
- DI-19708 - Notification Form

### CODE REVIEW

File: client-side/src/app/components/notification-block/notification-block.component.html [hunk 1]
```diff
@@ -1,14 +1,21 @@
...
   [noDataFoundMsg]= noDataMessage>
   <div class="right-buttons"  right-buttons>
-    <!-- <pep-button [sizeType]="'sm'" [value]= "'Add_Button' | translate" [classNames]="'pep-spacing-element'"  [styleType]="'strong'" (buttonClick)="addRelatedItems()"></pep-button> -->
+    <!-- <pep-button 
+    class="pep-spacing-element"
+    [sizeType]="'sm'" 
+    [value]="'Add' | translate" 
+    [styleType]="'strong'" 
+    (buttonClick)="navigateToNotificationsForm('Add')"
+    ></pep-button> -->
```
Comment (Chasky-H) [inline client-side/src/app/components/notification-block/notification-block.component.html:18]: There will never be an add AFAIK.

File: client-side/src/app/components/notification-block/notification-block.component.ts [hunk 2]
```diff
@@ -15,9 +20,11 @@ export class NotificationBlockComponent implements OnInit {
     private translate: TranslateService,
     private notificationsService: NotificationsService,
     private addonService: AddonService,
-    ){
-      this.addonService.addonUUID = "95025423-9096-4a4f-a8cd-d0a17548e42e"
-     }
+    private dialogService: PepDialogService,
+    private genericListService: PepGenericListService
+  ) {
+    this.addonService.addonUUID = "95025423-9096-4a4f-a8cd-d0a17548e42e"
```
Comment (Chasky-H) [inline client-side/src/app/components/notification-block/notification-block.component.ts:26]: Is there any way of getting this from the block hostObject?

File: client-side/src/app/components/notification-block/notification-block.component.ts [hunk 3]
```diff
@@ -122,11 +129,19 @@ export class NotificationBlockComponent implements OnInit {
     get: async (data) => {
       const actions = [];
 
+      if (data.rows.length === 1 && data?.selectionType != 0) {
+        actions.push({
+          title: this.translate.instant('Edit'),
```
Comment (Chasky-H) [inline client-side/src/app/components/notification-block/notification-block.component.ts:134]: Should this be 'edit'?
I think view is more appropriate.

File: server-side/notifications.service.ts [hunk 4]
```diff
@@ -33,28 +34,72 @@ class NotificationsService {
...
         else {
-            throw new Error(`UserUUID is required`);
+            const errors = validation.errors.map(error => error.stack.replace("instance.", ""));
+            throw new Error(errors.join("\n"));
+        }
+    }
+
+    // createas notifications by list of user emails, Subject and Body
+    async createNotifications(body){
+        let createdNotifications: AddonData[] = [];
```
Comment (simcha-waintrob) [inline server-side/notifications.service.ts:56]: It is better to replace 'body' with users, to clarify.
and also, all of the 'body'ies in the service.

**Review by simcha-waintrob — APPROVED** [review]

**Pull request merged**